src/encoder.cpp

```diff
@@ -77,5 +77,18 @@ void AgentHttpEncoder::handleResponse(const std::string& response) {
   }
 }
 
+bool AgentHttpEncoder::sampled(const ot::SpanContext& ctx, bool fallback_decision) try {
+  auto context = dynamic_cast<const SpanContext*>(&ctx);
```

Comment:
Looking at other code in the repo - looks like `dynamic_cast` is necessary here.

Reply:
Yup, it needs access a method that isn't in opentracing's `SpanContext`.